### PR TITLE
Base_oM: add `IContainer` interface

### LINKED ISSUE #1466: Base_oM: Create `IContainer` interface
#### Description:

As discussed with @IsakNaslundBh and @al-fisher, we need a new container interface to indicate classes that can contain several related objects bundled together whilst keeping them in separate properties. This will allow to define what are classically referred to as "models", i.e. sets of objects that are interrelated, e.g. representing a building, without imposing any requirement on the individual objects to coexist in the container.

This will allow to avoid the following kind of situations:
image

by replacing the "explode" with a BHoM_Engine method that "unpacks" the container and its properties in a flat list of objects. With reference to the picture, the `FEMmodel` should be an `IContainer` and should be unpackable via an `Unpack()` method, whose call can then be embedded in the BHoM_Adapter or other contexts.

### PR DESCRIPTION
**Pull request by alelom**
### Issues addressed by this PR

Closes #1466

### CODE REVIEW

**Review by IsakNaslundBh — APPROVED** [review]
Happy with this.

Interface implementing relevant IObject interface, and description makes sense.

File: BHoM/Interface/IContainer.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline BHoM/Interface/IContainer.cs:3]:
```suggestion
 * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
```

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: installer, versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core
@BHoMBot check serialisation

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Review by FraserGreenroyd — APPROVED** [review]
Approving based on @IsakNaslundBh previous review - only copyright changed.

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: ready-to-merge

**Pull request merged**